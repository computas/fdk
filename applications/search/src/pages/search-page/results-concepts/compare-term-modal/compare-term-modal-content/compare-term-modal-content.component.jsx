import React from 'react';
import PropTypes from 'prop-types';
import _get from 'lodash/get';
import _capitalize from 'lodash/capitalize';

import { getTranslateText } from '../../../../../lib/translateText';
import localization from '../../../../../lib/localization';

export const CompareTermModalContent = props => {
  let { terms, cols } = props;
  terms =
    terms && terms.length > 0
      ? terms
      : CompareTermModalContent.defaultProps.terms;
  cols = cols || CompareTermModalContent.defaultProps.cols;

  const title = items =>
    items.map((item, index) => {
      const { creator } = item;

      const publisherPrefLabel =
        getTranslateText(_get(creator, ['prefLabel'])) ||
        _capitalize(_get(creator, 'name', ''));

      return (
        <div className={cols} key={`title-${index}${item.uri}`}>
          <h3>
            {item && getTranslateText(item.prefLabel)
              ? getTranslateText(item.prefLabel)
                  .charAt(0)
                  .toUpperCase() +
                getTranslateText(item.prefLabel)
                  .substring(1)
                  .toLowerCase()
              : ''}
          </h3>
          <h5>{publisherPrefLabel}</h5>
        </div>
      );
    });

  const definition = items =>
    items.map((item, index) => (
      <div className={cols} key={`definition-${index}${item.uri}`}>
        <h5>{localization.compare.definition}</h5>
        <p>
          {getTranslateText(item.definition)
            ? getTranslateText(item.definition)
                .charAt(0)
                .toUpperCase() + getTranslateText(item.definition).substring(1)
            : ''}
        </p>
      </div>
    ));

  const inSchemeList = [];
  for (let i = 0; i < terms.length; i += 1) {
    let inScheme = '';
    if (terms[i].inScheme) {
      for (let j = 0; j < terms[i].inScheme.length; j += 1) {
        inScheme += terms[i].inScheme[j].substring(
          terms[i].inScheme[j].lastIndexOf('/') + 1
        );
        if (j < terms[i].inScheme.length - 1) {
          inScheme += ', ';
        }
      }
      inScheme += '.';
    }
    inSchemeList.push(inScheme);
  }

  const inScheme = items =>
    items.map((item, index) => (
      <div className={cols} key={`inScheme-${index}${item.uri}`}>
        <h5>{localization.compare.inScheme}</h5>
        <p>{item}</p>
      </div>
    ));

  const source = items =>
    items.map((item, index) => (
      <div className={cols} key={`source-${index}${item.uri}`}>
        <h5>{localization.compare.source}</h5>
        <p>
          {item.source ? (
            <a href={item.source}>
              {item.source}
              &nbsp;
              <i className="fa fa-external-link" />
            </a>
          ) : (
            ''
          )}
        </p>
      </div>
    ));

  const note = items =>
    items.map((item, index) => (
      <div className={cols} key={`note-${index}${item.uri}`}>
        <h5>{localization.compare.note}</h5>
        <p>{getTranslateText(item.note)}</p>
      </div>
    ));

  const altLabelList = [];
  for (let i = 0; i < terms.length; i += 1) {
    let altLabel = '';
    if (terms[i].altLabel) {
      for (let j = 0; j < terms[i].altLabel.length; j += 1) {
        if (getTranslateText(terms[i].altLabel[j])) {
          altLabel += getTranslateText(terms[i].altLabel[j]);
          if (j < terms[i].altLabel.length - 1) {
            altLabel += ', ';
          }
        }
      }
      altLabel += '.';
    }
    altLabelList.push(altLabel);
  }

  const altLabel = items =>
    items.map((item, index) => (
      <div className={cols} key={`altLabel-${index}${item.uri}`}>
        <h5>{localization.compare.altLabel}</h5>
        <p>{item}</p>
      </div>
    ));

  return (
    <div>
      <div className="row">{title(terms)}</div>
      <hr />
      <div className="row">{definition(terms)}</div>
      <hr />
      <div className="row">{inScheme(inSchemeList)}</div>
      <hr />
      <div className="row">{source(terms)}</div>
      <hr />
      <div className="row">{note(terms)}</div>
      <hr />
      <div className="row">{altLabel(altLabelList)}</div>
    </div>
  );
};

CompareTermModalContent.defaultProps = {
  terms: [
    {
      uri: '',
      identifier: '',
      prefLabel: {
        nb: ''
      },
      definition: {
        nb: ''
      },
      note: {
        nb: ''
      },
      source: '',
      creator: {
        name: ''
      },
      inScheme: ['']
    }
  ],
  cols: 'col-md-6'
};

CompareTermModalContent.propTypes = {
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      uri: PropTypes.string,
      identifier: PropTypes.string,
      prefLabel: PropTypes.shape({}),
      definition: PropTypes.shape({}),
      note: PropTypes.shape({}),
      source: PropTypes.string,
      creator: PropTypes.shape({}),
      inScheme: PropTypes.array
    })
  ),
  cols: PropTypes.string
};
