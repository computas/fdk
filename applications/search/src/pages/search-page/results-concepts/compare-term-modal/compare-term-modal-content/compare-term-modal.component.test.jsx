import React from 'react';
import { shallow } from 'enzyme';
import { CompareTermModalContent } from './compare-term-modal-content.component';
import concepts from '../../../../../../test/fixtures/concepts';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    terms: [concepts.hits.hits[0]._source, concepts.hits.hits[1]._source],
    cols: '6'
  };
  wrapper = shallow(<CompareTermModalContent {...defaultProps} />);
});

test('should render CompareTermModalContent correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
