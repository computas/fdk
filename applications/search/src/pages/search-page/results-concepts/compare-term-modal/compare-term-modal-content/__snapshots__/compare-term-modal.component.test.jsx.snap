// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CompareTermModalContent correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="6"
      key="title-0https://data-david.github.io/Begrep/begrep/Enhet"
    >
      <h3>
        Enhet
      </h3>
      <h5>
        Brønnøysundregistrene
      </h5>
    </div>
    <div
      className="6"
      key="title-1https://data-david.github.io/Begrep/begrep/Organisasjonsnummer"
    >
      <h3>
        Organisasjonsnummer
      </h3>
      <h5 />
    </div>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="6"
      key="definition-0https://data-david.github.io/Begrep/begrep/Enhet"
    >
      <h5>
        Definition
      </h5>
      <p>
        Alt som er registrert med et organisasjonsnummer 
      </p>
    </div>
    <div
      className="6"
      key="definition-1https://data-david.github.io/Begrep/begrep/Organisasjonsnummer"
    >
      <h5>
        Definition
      </h5>
      <p>
        Et nisifret nummer som tildeles av Enhetsregisteret for en organisasjon som skal operere som en offentlig aktør
      </p>
    </div>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="6"
      key="inScheme-0undefined"
    >
      <h5>
        Domain / application
      </h5>
      <p>
        Befolkning.
      </p>
    </div>
    <div
      className="6"
      key="inScheme-1undefined"
    >
      <h5>
        Domain / application
      </h5>
      <p />
    </div>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="6"
      key="source-0https://data-david.github.io/Begrep/begrep/Enhet"
    >
      <h5>
        Source
      </h5>
      <p>
        <a
          href="https://jira.brreg.no/browse/BEGREP-208"
        >
          https://jira.brreg.no/browse/BEGREP-208
           
          <i
            className="fa fa-external-link"
          />
        </a>
      </p>
    </div>
    <div
      className="6"
      key="source-1https://data-david.github.io/Begrep/begrep/Organisasjonsnummer"
    >
      <h5>
        Source
      </h5>
      <p />
    </div>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="6"
      key="note-0https://data-david.github.io/Begrep/begrep/Enhet"
    >
      <h5>
        Description
      </h5>
      <p>
        Alle hovedenheter, underenheter og organisasjonsledd som er identifisert med et organisasjonsnummer.
      </p>
    </div>
    <div
      className="6"
      key="note-1https://data-david.github.io/Begrep/begrep/Organisasjonsnummer"
    >
      <h5>
        Description
      </h5>
      <p />
    </div>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="6"
      key="altLabel-0undefined"
    >
      <h5>
        Also known as:
      </h5>
      <p />
    </div>
    <div
      className="6"
      key="altLabel-1undefined"
    >
      <h5>
        Also known as:
      </h5>
      <p>
        Orgid, Orgnr.
      </p>
    </div>
  </div>
</div>
`;
